src/core/Akka.Persistence.Query/Offset.cs

```diff
@@ -0,0 +1,115 @@
+        /// as the `offset` parameter in a subsequent query.
+        /// </summary>
+        public static Offset Sequence(long value) => new Sequence(value);
+
+        /// <summary>
+        /// Corresponds to an ordered unique identifier of the events. Note that the corresponding
+        /// offset of each event is provided in the <see cref="EventEnvelope"/>,
+        /// which makes it possible to resume the stream at a later point from a given offset.
+        /// The `offset` is exclusive, i.e.the event with the exact same sequence number will not be included
+        /// in the returned stream.This means that you can use the offset that is returned in <see cref="EventEnvelope"/>
+        /// as the `offset` parameter in a subsequent query.
+        /// </summary>
+        public static Offset TimeBasedGuid(Guid value) => new TimeBasedGuid(value);
+    }
+
+    /// <summary>
+    /// Corresponds to an ordered sequence number for the events.Note that the corresponding
+    /// offset of each event is provided in the <see cref="EventEnvelope"/>,
+    /// which makes it possible to resume the stream at a later point from a given offset.
+    /// The `offset` is exclusive, i.e.the event with the exact same sequence number will not be included
+    /// in the returned stream. This means that you can use the offset that is returned in <see cref="EventEnvelope"/>
+    /// as the `offset` parameter in a subsequent query.
+    /// </summary>
+    public sealed class Sequence : Offset
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Sequence"/> class.
+        /// </summary>
+        public Sequence(long value)
+        {
+            Value = value;
+        }
+
+        public long Value { get; }
+
+        private bool Equals(Sequence other) => Value == other.Value;
+
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(null, obj)) return false;
+            if (ReferenceEquals(this, obj)) return true;
+            return obj is Sequence && Equals((Sequence)obj);
+        }
+
+        public override int GetHashCode() => Value.GetHashCode();
+    }
+
+    /// <summary>
+    /// Corresponds to an ordered unique identifier of the events. Note that the corresponding
+    /// offset of each event is provided in the <see cref="EventEnvelope"/>,
+    /// which makes it possible to resume the stream at a later point from a given offset.
+    /// The `offset` is exclusive, i.e.the event with the exact same sequence number will not be included
+    /// in the returned stream.This means that you can use the offset that is returned in <see cref="EventEnvelope"/>
+    /// as the `offset` parameter in a subsequent query.
+    /// </summary>
+    public sealed class TimeBasedGuid : Offset
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TimeBasedGuid"/> class.
+        /// </summary>
+        public TimeBasedGuid(Guid value)
```

Comment:
@alexvaluyskiy it is, but it can be safely and implicitly cast to a `Guid`: http://docs.datastax.com/en/drivers/csharp/3.3/html/T_Cassandra_TimeUuid.htm